R/coh_plot.R

```diff
@@ -0,0 +1,324 @@
+#' @param cumulative if TRUE returns cumulative Hazards (1-Survival)
+#' @return ggplot2 plot of survival/cumulative hazard
+#' @examples
+#' # load example package data
+#' data("cohortdata")
+#' head(cohortdata)
+#' start_cohort <- as.Date("2044-01-01")
+#' end_cohort <- as.Date("2044-12-31")
+#'
+#' cohortdata$immunization_death <-
+#'   get_immunization_date(
+#'     data = cohortdata,
+#'     outcome_date_col = "death_date",
+#'     outcome_delay = 0,
+#'     immunization_delay = 14,
+#'     vacc_date_col = c("vaccine_date_1", "vaccine_date_2"),
+#'     end_cohort = end_cohort,
+#'     take_first = FALSE
+#'   )
+#'
+#' cohortdata$vaccine_status <- set_status(
+#'   data = cohortdata,
+#'   col_names = "immunization_death",
+#'   status = c("v", "u")
+#' )
+#'
+#' cohortdata$death_status <- set_status(
+#'   data = cohortdata,
+#'   col_names = "death_date"
+#' )
+#'
+#' cohortdata$time_to_death <- get_time_to_event(
+#'   data = cohortdata,
+#'   outcome_date_col = "death_date",
+#'   start_cohort = start_cohort,
+#'   end_cohort = end_cohort,
+#'   start_from_immunization = FALSE
+#' )
+#'
+#' plot_survival(data = cohortdata,
+#'   outcome_status_col = "death_status",
+#'   time_to_event_col = "time_to_death",
+#'   vacc_status_col = "vaccine_status",
+#'   vaccinated_status = "v",
+#'   unvaccinated_status = "u",
+#'   vaccinated_color = "steelblue",
+#'   unvaccinated_color = "darkred",
+#'   start_cohort = start_cohort,
+#'   end_cohort = end_cohort,
+#'   percentage = TRUE,
+#'   cumulative = TRUE
+#' )
+#' @export
+plot_survival <- function(data, outcome_status_col,
+                          time_to_event_col,
+                          vacc_status_col,
+                          vaccinated_status = "v",
+                          unvaccinated_status = "u",
+                          vaccinated_color = "steelblue",
+                          unvaccinated_color = "darkred",
+                          start_cohort,
+                          end_cohort,
+                          percentage = TRUE,
+                          cumulative = FALSE) {
+  # input checking
+  checkmate::assert_data_frame(
+    data,
+    min.rows = 1L
+  )
+  checkmate::assert_names(
+    names(data),
+    must.include = c(outcome_status_col, time_to_event_col, vacc_status_col)
+  )
+  checkmate::assert_character(
+    c(vaccinated_color, unvaccinated_color)
+  )
+  checkmate::assert_logical(
+    percentage, len = 1
+  )
+  checkmate::assert_logical(
+    cumulative, len = 1
+  )
+  # check date types
+  checkmate::assert_date(
+    start_cohort, any.missing = FALSE, len = 1
+  )
+  checkmate::assert_date(
+    end_cohort, any.missing = FALSE, len = 1
+  )
+  checkmate::assert_names(
+    data[[vacc_status_col]],
+    must.include = c(vaccinated_status, unvaccinated_status)
+  )
+
+  #KM model
+  km_model <- survival::survfit(
+    survival::Surv(
+      data[[time_to_event_col]],
+      data[[outcome_status_col]]
+    ) ~ data[[vacc_status_col]]
+  )
+
```

Comment:
I'm not sure if survfit always returns the strata in the same order. That's why I decided extracting them using the name of the strata based on the model description:
`conn1 <- paste0("data[[vacc_status_col]]=", vaccinated_status)
  results_1 <- results[results$strata == conn1, ]
  conn2 <- paste0("data[[vacc_status_col]]=", unvaccinated_status)
  results_2 <- results[results$strata == conn2, ]`
In that way I'm always sure that results_1 contains the results for vaccinated and results_2 the ones for unvaccinated.

Reply:
Should I define as factors the levels of the columns created in the set_status() function to always be sure about the orders?

Reply:
For now, I will solve it here:
`results$strata <- factor(results$strata,`
  `levels = c(paste0("data[[vacc_status_col]]=", vaccinated_status),`
                   `paste0("data[[vacc_status_col]]=", unvaccinated_status))`
`)`
`levels(results$strata) <- c(vaccinated_status, unvaccinated_status)`
To avoid changing the behavior of the package before the first test. But, in order to make clear the values of vaccinated_status, unvaccinated_status, and event_status for users and avoid unexpected behaviors from survival, in the next PR, I'm going create functions to generate the vaccine_status  and the event status separately using `set_status()`, as @jpavlich suggested.

Reply:
With this partial solution, I can accept the simplification of the plotting below.

Reply:
Resolved in https://github.com/epiverse-trace/vaccineff/pull/31/commits/304d016dc2f8cf5e8b1896001213d7c87975e6bc